source/develop/release-management/features/network/sriov-live-migration.md

```diff
@@ -0,0 +1,178 @@
+that required additional configuration inside affected VM.
+
+The state of the feature is tracked in [[RFE] [SR-IOV] Migration should not require downtime as of today][2].
+Related patches can be found on [gerrit topic:sriov_live_migration][3].
+
+## Owner
+
+*   Name: [Ales Musil](https://github.com/almusil)
+
+*   Email: <amusil@redhat.com>
+
+
+## Detailed Description
+
+In order to migrate a VM that is connected to SR-IOV passthrough profile
+we need to first unplug the SR-IOV device from a VM on the source, migrate
+the VM and plug SR-IOV device on the destination. This causes longer downtime
+during which the VM is not connected to the network.
+
+The current implementation allowed to mitigate this downtime
+[Live Migration Support For SRIOV][1], however this approach required manual
+configuration to be done in the guest which is not convenient for large setups. 
+
+In order to achieve minimal downtime the [NET_FAILOVER][4] support was added 
+to Linux kernel. The driver provides automated mechanism which is transparent 
+to the VM, and the user space. It is achieved by `failover` netdev which acts
+as master device and controls two slave devices, the `primary` device being 
+SR-IOV passtrough and the virtio as `standby`. The VM user access network
+through the `failover` device. 
+
+There is still need to provide two interfaces, the SR-IOV passtrough device
+and virtio device, to the affected VM, but the  overall configuration overhead
+has been reduced to selecting suitable vNIC profile pair.
+
+
+## Prerequisites
+
+The implementation requires a [Linux kernel][5], [libvirt][6] and [qemu][7] 
+support with `net_failover`. 
+
+The [Linux kernel][5] support has to be only on guest side. 
+
+## Limitations
+
+* In addition to the SR-IOV NIC, a linux-bridge based (or an OVN physical network in OVS clusters)
+  has to be connected to the source, and the destination host.
+* Because of the required support in the guest, only RHEL8 guests are supported (Linux kernel version >=4.18 or >=5.10).
+
+## Implementation
+
+### Entity Description
+
+vNIC profile will get a new attribute that will point to the failover virtio 
+vNIC profile. This selection will be only available with `Passthrough` and `Migratable`
+selected. 
+
+### User Experience
+
+Upon creation of vNIC profile, the user can enable failover and select 
+the failover network vNIC profile. The `Failover` selection is available 
+only for `Passthrough` and `Migratable` vNIC profiles.
+
+REST API takes an additional parameter for vNIC called `failover`
+containing id of the failover vNIC profile:
+
+```
+<vnic_profile id="123">
+  <name>vNIC_name</name>
+  <pass_through>
+    <mode>enabled</mode>
+  </pass_through>
+  <migratable>true</migratable>
+  <failover id="456"/>
+</vnic_profile>
+```
+
+VM interface then requires only single vNIC profile that has the `failover` attribute 
+in order to utilize this behavior.
+
+### Engine backend
+
```

Comment:
- I understood that the live-migration code has to be modified, to skip the unplug of the SR-IOV NIC. If this, this should be mentioned here.
- I also understood, that we should have a new cluster level to ensure that all hosts in the cluster have the required libvirt and qemu versions.

Reply:
Will this break migration implementation and logic that we currently have? will this mean that the Vf won't get hotunplugged from the source host?

Reply:
Also, about the required network. We agreed that making the fail-over network as required, is not the right direction and might introduce issues.

Reply:
The required statement is not there anymore.

Reply:
I have mentioned the skip of unplug and the cluster level version